__tests__/components/templates/SinopiaResourceTemplates.test.js

```diff
@@ -1,62 +1,93 @@
+import { createMemoryHistory } from 'history'
+/* eslint import/no-unresolved: 'off' */
+import { renderWithReduxAndRouter, createReduxStore } from 'testUtils'
+import { saveAs } from 'file-saver'
 
-jest.mock('sinopiaServer')
-
-describe('<SinopiaResourceTemplates />', () => {
-  const messages = [
-    'Created http://localhost:8080/repository/ld4p/Note/sinopia:resourceTemplate:bf2:Note1',
-    'Created http://localhost:8080/repository/ld4p/Note/sinopia:resourceTemplate:bf2:Note2',
-  ]
-
-  const resourceTemplateSummary = {
-    name: 'Note',
-    key: 'ld4p:resourceTemplate:bf2:Note',
-    id: 'ld4p:resourceTemplate:bf2:Note',
-    author: 'wright.lee.renønd',
-    remark: 'very salient information',
-    group: 'stanford',
+jest.mock('file-saver')
+
+const createInitialState = () => {
+  const state = {
+    selectorReducer: {
+      resource: {
+        'resourceTemplate:bf2:Note': {
+          'http://id.loc.gov/ontologies/bibframe/note': {
+            items: {},
+          },
+        },
+      },
+      entities: {
+        resourceTemplateSummaries,
+        resourceTemplates: {},
+      },
+      editor: {
+        serverError: '',
+      },
+    },
   }
+  return state
+}
 
-  const resourceTemplateSummaries = [resourceTemplateSummary]
+const resourceTemplateSummary = {
+  name: 'Note',
+  key: 'resourceTemplate:bf2:Note',
+  id: 'resourceTemplate:bf2:Note',
+  author: 'wright.lee.renønd',
+  remark: 'very salient information',
+  group: 'ld4p',
+}
 
-  const wrapper = shallow(<SinopiaResourceTemplates.WrappedComponent messages={messages} resourceTemplateSummaries={resourceTemplateSummaries} />)
+const resourceTemplateSummaries = [resourceTemplateSummary]
 
+describe('SinopiaResourceTemplates', () => {
   it('has a header for the area where the table of resource templates for the groups are displayed', () => {
-    expect(wrapper.find('div > h4').last().text()).toEqual('Available Resource Templates in Sinopia')
+    const store = createReduxStore(createInitialState())
+    const { getByText } = renderWithReduxAndRouter(
+      <SinopiaResourceTemplates messages={[]} />, store,
+    )
+
+    expect(getByText('Available Resource Templates in Sinopia')).toBeInTheDocument()
   })
 
   it('has a bootstrap table that displays the results from the calls to sinopia_server', () => {
-    expect(wrapper.find('BootstrapTableContainer').length).toEqual(1)
+    const store = createReduxStore(createInitialState())
+    const { container, getByText, getAllByText } = renderWithReduxAndRouter(
+      <SinopiaResourceTemplates messages={[]} />, store,
+    )
+
+    expect(container.querySelector('table#resource-template-list')).toBeInTheDocument()
+    expect(getByText(/Template name/)).toBeInTheDocument()
+    expect(getByText(/ID/)).toBeInTheDocument()
+    expect(getByText(/Author/)).toBeInTheDocument()
+    expect(getByText(/Guiding statement/)).toBeInTheDocument()
+    expect(getAllByText(/Download/)[0]).toBeInTheDocument()
```

Comment:
Unclear what the `[0]` is intended to test.

Reply:
There is the text "download" for the header, and the link "download" for the download. Because there are two of them I needed to use getAllByText, and it is an array. I will add a comment.

Reply:
There is the text "download" for the header, and the link "download" for the download. Because there are two of them I needed to use getAllByText, and it is an array. "Download" in the header is the first one in the array.

Reply:
Then add a test-id: https://testing-library.com/docs/dom-testing-library/api-queries#bytestid